Correct the signature of `FloatingPoint.ldexp` (#5772)
ponylang/ponyc: .release-notes/correct-ldexp-signature.md

## Correct the signature of `FloatingPoint.ldexp`

> `FloatingPoint.ldexp` scales a floating-point value by a power of two (the inverse of `frexp`), but its signature took the value to scale as an explicit first argument and ignored the receiver it was called on. This was inconsistent with the rest of the `FloatingPoint` API, where methods operate on the receiver.

Reviewer: Two things: "ignored", and "the rest of the `FloatingPoint` API" — nine members declared on the trait don't operate on the receiver.

```suggestion
`FloatingPoint.ldexp` scales a floating-point value by a power of two (the inverse of `frexp`), but it took the value to scale as an explicit first argument and never used the receiver it was called on. Operations on a float elsewhere in `FloatingPoint` work on the receiver.
```

> The redundant first parameter has been removed. `ldexp` now scales the value it is called on, so the receiver is no longer ignored.

Reviewer: The first parameter wasn't the redundant one, the receiver was. As written this sends someone to `F64(0).ldexp(exponent)`, which compiles and returns `0`. The tail restates the first half of the sentence, so it goes too.

```suggestion
The value to scale has moved from the first argument to the receiver. `ldexp` now scales the value it is called on.
```

Before:

```pony

> // the receiver (here F64(0)) was ignored; `mantissa` was the value scaled

Reviewer: Same as line 3.

```suggestion
// the receiver (here F64(0)) was never used; `mantissa` was the value scaled
```
